### PR TITLE
Add String#to_utf16 and String.from_utf16

### PR DESCRIPTION
**Pull request by asterite**
This will be useful to interface with Windows functions that expect/return UTF-16 values.

### CODE REVIEW

File: src/string/utf16.cr [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    i = 0
+    each_char do |char|
+      ord = char.ord
+      if ord <= 0xd800 || (0xe000 <= ord < 0x10000)
+        # One UInt16 is enough
+        slice[i] = ord.to_u16
+      elsif ord >= 0x10000
+        # Needs surrogate pair
+        ord -= 0x10000
+        slice[i] = 0xd800_u16 + ((ord >> 10) & 0x3ff) # Keep top 10 bytes
```
Comment (straight-shoota) [inline src/string/utf16.cr]: `top 10 bits`?

Comment (asterite) [reply src/string/utf16.cr]: Thanks, fixed (did amend)

File: spec/std/string/utf16.cr [hunk 2]
```diff
@@ -0,0 +1,47 @@
+require "spec"
```
Comment (RX14) [inline spec/std/string/utf16.cr]: `utf16_spec.cr`

Comment (asterite) [reply spec/std/string/utf16.cr]: fixed, thanks

File: src/string/utf16.cr [hunk 3]
```diff
@@ -0,0 +1,93 @@
+class String
+  # Returns the UTF-16 encoding of the given *string*.
+  #
+  # Invalid chars (in the range U+D800..U+DFFF) are encoded with the
+  # unicode replacement char value `0xfffd`.
+  #
+  # ```
+  # UTF16.encode("hi 𐂥") # => Slice[104_u16, 105_u16, 32_u16, 55296_u16, 56485_u16]
```
Comment (Sija) [inline src/string/utf16.cr]: `UTF16.encode("hi 𐂥")` => `"hi 𐂥".to_utf16`

Comment (asterite) [reply src/string/utf16.cr]: thanks, I forgot to update the docs

**Pull request merged**